pywt/_extensions/c/cwt.template.c

```diff
@@ -0,0 +1,267 @@
+/* Copyright (c) 2016 Holger Nahrstaedt */
+/* See COPYING for license details. */
+
+#include "templating.h"
+
+#ifndef TYPE
+#error TYPE must be defined here.
+#else
+
+#include "cwt.h"
+#include "convolution.h"
+
+#if defined _MSC_VER
+#define restrict __restrict
+#elif defined __GNUC__
+#define restrict __restrict__
+#endif
+#define _USE_MATH_DEFINES
+#include "math.h"
+
+TYPE CAT(TYPE, _powof)(const TYPE x, const TYPE y)
+{
+    if (sizeof(TYPE) == sizeof(double))
+        return pow(x,y);        
+    else
+        return powf(x,y);
+}
+
+
+
+void CAT(TYPE, _gaus)(const TYPE * const restrict input,
+                              TYPE * const restrict output, const size_t N,
+                              const size_t number){
+    size_t i = 0;
+    for (i = 0; i < N; i++)
+    {
+           switch (number) {
+               case 1:
+                    CAT(TYPE, _gaus1)(input[i], &output[i]);
+                    break;
+                case 2:
+                    CAT(TYPE, _gaus2)(input[i], &output[i]);
+                    break;
+                case 3:
+                    CAT(TYPE, _gaus3)(input[i], &output[i]);
+                    break;
+                case 4:
+                    CAT(TYPE, _gaus4)(input[i], &output[i]);
+                    break;
+                case 5:
+                    CAT(TYPE, _gaus5)(input[i], &output[i]);
+                    break;
+                case 6:
+                    CAT(TYPE, _gaus6)(input[i], &output[i]);
+                    break;
+                case 7:
+                    CAT(TYPE, _gaus7)(input[i], &output[i]);
+                    break;
+                case 8:
+                    CAT(TYPE, _gaus8)(input[i], &output[i]);
+                    break;
+          }
+    }
+}
+
+void CAT(TYPE, _gaus1)(const TYPE  input, TYPE * const restrict output)
```

Comment:
These aren't used anywhere else, so I would inline them into `TYPE_gaus` (unless I'm missing something?) It'd make things a bit more compact.